Prevent chests connecting on claim boundary

FrankHeijden opened the pull request:
If a chest is placed on the claim boundary, thieves can place a chest connecting to the chest in the claim and thus loot the contents. This PR fixes that bug.

FrankHeijden committed:
Prevent chests connecting on claim boundary

Jikoo commented on src/main/java/me/ryanhamshire/GriefPrevention/BlockEventHandler.java:
```diff
@@ -285,6 +293,36 @@ public void onBlockPlace(BlockPlaceEvent placeEvent)
+                Block relative = block.getRelative(face);
+                if (!(relative.getBlockData() instanceof Chest)) continue;
+
+                Claim relativeClaim = this.dataStore.getClaimAt(relative.getLocation(), true, playerData.lastClaim);
```
Should use `claim` here instead of `playerData.lastClaim`, odds are much higher that chest is placed in same claim as an adjacent block.

FrankHeijden replied:
Added!

FrankHeijden committed:
Check relative claim with the claim the other connected component might be in

FrankHeijden committed:
Merge branch 'master' into fix/chests

FrankHeijden committed:
Allow chests to connect in claims of same owner

PaulBGD commented on src/main/java/me/ryanhamshire/GriefPrevention/BlockEventHandler.java:
```diff
@@ -285,6 +293,38 @@ public void onBlockPlace(BlockPlaceEvent placeEvent)
+            for (BlockFace face : HORIZONTAL_DIRECTIONS)
+            {
+                Block relative = block.getRelative(face);
+                if (!(relative.getBlockData() instanceof Chest)) continue;
```
Ternary

PaulBGD commented on src/main/java/me/ryanhamshire/GriefPrevention/BlockEventHandler.java:
```diff
@@ -285,6 +293,38 @@ public void onBlockPlace(BlockPlaceEvent placeEvent)
+
+                // Chests outside claims should connect (both null)
+                // and chests inside the same claim should connect (equal)
+                if (Objects.equals(claimOwner, relativeClaimOwner)) break;
```
Ternary

PaulBGD reviewed (changes requested):
Ternary

RoboMWM committed:
refactor doublechest across claim boundary check to a separate method

RoboMWM committed:
expand shorthand if-else

thought I did this but I guess in the midst of trying to push this to
the right branch I somehow ended up not doing this

RoboMWM committed:
Merge branch 'master' into fix/chests
